h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderModel.java

```diff
@@ -463,30 +513,45 @@ protected void removeNumeratorAndDenominatorColumns(Frame fr) {
   private static class KFoldEncodingStrategy extends EncodingStrategy {
 
     String _foldColumn;
+    int _outOfFold;
     
-    public KFoldEncodingStrategy(String foldColumn, 
+    public KFoldEncodingStrategy(String foldColumn, int outOfFold,
                                  BlendingParams blendingParams, double noise, long seed) {
       super(blendingParams, noise, seed);
       _foldColumn = foldColumn;
+      _outOfFold = outOfFold;
     }
 
     @Override
     public Frame doApply(Frame fr, String columnToEncode, Frame encodings, String encodedColumn, int targetClass) {
+      Frame workingFrame = fr;
       int teColumnIdx = fr.find(columnToEncode);
-      int foldColIdx = fr.find(_foldColumn);
+      int foldColIdx;
+      if (_outOfFold== NO_FOLD) {
+        foldColIdx = fr.find(_foldColumn);
+      } else {
+        workingFrame = new Frame(fr);
+        Vec tmpFoldCol = workingFrame.anyVec().makeCon(_outOfFold);
+        Scope.track(tmpFoldCol);
+        workingFrame.add(new String[] {_foldColumn+TMP_COLUMN_POSTFIX}, new Vec[]{tmpFoldCol});
+        foldColIdx = workingFrame.numCols()-1;
+      }
       int encodingsFoldColIdx = encodings.find(_foldColumn);
       int encodingsTEColIdx = encodings.find(columnToEncode);
       long[] foldValues = getUniqueColumnValues(encodings, encodingsFoldColIdx);
       int maxFoldValue = (int) ArrayUtils.maxValue(foldValues);
-      double priorMean = computePriorMean(encodings);
+      double priorMean = computePriorMean(encodings); //FIXME: we want prior for the outOfFold encodings 
```

Comment:
not fixing this now because the whole TE implementation is currently using a global prior with KFold strategy.
It probably creates a small data leakage, but as it is a legacy issue, will fix in a future improvement.